src/main/java/book/store/service/impl/CartItemServiceImpl.java

```diff
@@ -16,13 +19,29 @@
 public class CartItemServiceImpl implements CartItemService {
     private final CartItemRepository cartItemRepository;
     private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
     private final CartItemMapper cartItemMapper;
 
     @Override
     public void create(String username, CartItemCreateRequestDto requestDto) {
-        CartItem cartItem = cartItemMapper.toModel(requestDto);
-        cartItem.setShoppingCart(shoppingCartRepository.findShoppingCartByUserEmail(username));
-        cartItemRepository.save(cartItem);
+        Optional<CartItem> cartItemOptional = isBookPresentInShoppingCart(username, requestDto);
+        if (cartItemOptional.isEmpty()) {
+            //if the book is in the shopping cart,
```

Comment:
Instead of leaving a comment, create private methods with appropriate names and move the logic there so that it is clear what is being done